packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -58,7 +59,18 @@ export const RegistrationForm = ({
 }: RegistrationFormProps): ReactElement => {
   const { trackEvent } = useContext(AnalyticsContext);
   const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [username, setUsername] = useState('');
+  const [name, setName] = useState('');
   const isAuthorOnboarding = trigger === AuthTriggers.Author;
+  const { data: usernameData, isLoading } = useGenerateUsername(name);
+
+  useEffect(() => {
+    if (username.length || isLoading) return;
+
+    if (usernameData?.generateUniqueUsername) {
+      setUsername(usernameData.generateUniqueUsername);
+    }
+  }, [usernameData, isLoading, username, setUsername]);
```

Comment:
Was it not possible to add this inside the hook rather as it's basically the same code twice now?